src/core/agent/services/identifierService.ts

```diff
@@ -208,6 +208,10 @@ class IdentifierService extends AgentService {
       isDeleted: true,
     });
 
+    await this.props.signifyClient.identifiers().update(identifier, {
+      name: `XX-${randomSalt()}:${metadata.displayName}`,
```

Comment:
Actually @Sotatek-BaoHoanga we should update this before setting `isDeleted: true`. So that if it crashes right before this line, it'll get completed again on next restart with your other work on pending deletion.